src/boot/efi/console.c

```diff
@@ -25,49 +43,74 @@ EFI_STATUS console_key_read(UINT64 *key, BOOLEAN wait) {
                 if (EFI_ERROR(err))
                         TextInputEx = NULL;
 
+                err = uefi_call_wrapper(BS->CheckEvent, 1, TextInputEx->WaitForKeyEx);
```

Comment:
if TextIputEx is NULL this will result in a segfault (i.e. if LibLocateProtocol failed)